src/odemis/driver/test/xt_client_test.py

```diff
@@ -712,6 +712,120 @@ def test_apply_autofocus(self):
+        self.assertTrue("unit" in mpp_orientation_info)
+        self.assertIsInstance(mpp_orientation_info["range"], list)
+        self.assertEqual(len(mpp_orientation_info["range"]), 2)
+
+    def test_get_aperture_index(self):
+        """
+        Test getting the aperture index.
+        """
+        if self.xt_type != 'xttoolkit':
+            self.skipTest("This test needs XTToolkit to run.")
+
+        aperture_index = self.microscope.get_aperture_index()
+        self.assertIsInstance(aperture_index, int)
+        self.assertTrue(0 <= aperture_index <= 14)
+
+    def test_set_aperture_index(self):
+        """
+        Test setting the aperture index.
+        """
+        if self.xt_type != 'xttoolkit':
+            self.skipTest("This test needs XTToolkit to run.")
+
+        current_aperture_index = self.microscope.get_aperture_index()
+        aperture_range = tuple(self.microscope.aperture_index_info()["range"])
+        new_aperture_index = current_aperture_index + 1 if current_aperture_index < aperture_range[1] else \
+                                                                                             current_aperture_index - 1
+
+        self.microscope.set_aperture_index(new_aperture_index)
+        aperture_index = self.microscope.get_aperture_index()
+        self.assertEqual(aperture_index, new_aperture_index)
+
+        self.microscope.set_aperture_index(current_aperture_index)
+        aperture_index = self.microscope.get_aperture_index()
+        self.assertEqual(aperture_index, current_aperture_index)
+
+    def test_aperture_index_info(self):
+        """
+        Test getting the aperture index info.
+        """
+        if self.xt_type != 'xttoolkit':
+            self.skipTest("This test needs XTToolkit to run.")
+
+        aperture_index_info = self.microscope.aperture_index_info()
+        self.assertIsInstance(aperture_index_info, dict)
+        self.assertIsInstance(aperture_index_info["range"], list)
+        self.assertEqual(len(aperture_index_info["range"]), 2)
+
+    def test_get_beamlet_index(self):
+        """
+        Test getting the beamlet index.
+        """
+        if self.xt_type != 'xttoolkit':
+            self.skipTest("This test needs XTToolkit to run.")
+
+        beamlet_index = self.microscope.get_beamlet_index()
+        self.assertIsInstance(beamlet_index, tuple)
+        self.assertEqual(len(beamlet_index), 2)
+        self.assertTrue(1 <= beamlet_index[0] <= 8)
+        self.assertTrue(1 <= beamlet_index[1] <= 8)
+
+    def test_set_beamlet_index(self):
+        """
+        Test setting the beamlet index.
+        """
+        if self.xt_type != 'xttoolkit':
+            self.skipTest("This test needs XTToolkit to run.")
+
+        current_index = self.microscope.get_beamlet_index()
+        # Make sure both the x and the y value are changed.
+        new_beamlet_index = (3, 3) if 3 not in current_index else (5, 5) if 5 not in current_index else (6, 6)
```

Comment:
Is there a reason you use the same index in x and y? I'd tend to prefer different ones, in a test case, so that if for some reasons x and y are inverted, it's more likely you notice it. If yes, please a comment. If not, maybe just using either 3,5 or 3,6 would be fine?